sympy/core/numbers.py

```diff
@@ -252,6 +252,10 @@ def igcd(*args):
     if 1 in args_temp:
         return 1
     a = args_temp.pop()
+    if HAS_GMPY: # Using gmpy if present to speed up.
+        for b in args_temp:
+            a = as_int(gmpy.gcd(a, b)) if b else a
```

Comment:
It would be better to call as_int only once in the return line. Otherwise this converts back and forth between Python and gmpy integers each time the loop body is run.